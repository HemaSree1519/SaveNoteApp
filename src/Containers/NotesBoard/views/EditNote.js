import {Button, Input, Label, Modal, ModalBody, ModalFooter, ModalHeader} from "reactstrap";
import Moment from "moment";
import React, {Component} from "react";

export default class EditNote extends Component {
    render() {
        return (
            <Modal isOpen={this.props.childProps.isEditingNote} toggle={this.props.childProps.setEditState}>
                <ModalHeader>
                    <Input type="textarea"
                           name="title"
                           className="note-card-title-input"
                           value={this.props.childProps.editingNoteTitle}
                           onChange={this.props.childProps.onEditNoteTitle}
                    />
                </ModalHeader>
                <ModalBody>
                    <Input type="textarea"
                           name="content"
                           className="note-card-content-input"
                           value={this.props.childProps.editingNoteContent}
                           onChange={this.props.childProps.onEditNoteContent}
                           rows={5}/>
                    <br/>
                    <div className="footer-container">
                        <Label className="footer-edited-time">
                            Edited {Moment(this.props.childProps.editingNote['updatedAt']).format('lll')}</Label>
                    </div>
                </ModalBody>
                <ModalFooter className="modal-footer">
                    <Button name="delete" className="footer-button"
                            onClick={this.props.childProps.onDelete}>Delete</Button>{'  '}
                    <Button name="close" className="footer-button"
                            onClick={this.props.childProps.onUpdateNote}>Close</Button>
                </ModalFooter>
            </Modal>
        )
    }
};