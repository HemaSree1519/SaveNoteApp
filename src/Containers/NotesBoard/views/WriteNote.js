import {Button, Card, Input} from "reactstrap";
import React, {Component} from "react";

export default class WriteNote extends Component{
    render(){
        return (
            <Card className="note-input">
                <Input className="note-card-title-input" type="text" placeholder="Title" name="title"
                       value={this.props.childProps.writingNoteTitle}
                       onChange={this.props.childProps.onWriteNoteTitle}
                />
                <Input className="note-card-content-input" type="textarea" placeholder="Write a note..." name="content"
                       value={this.props.childProps.writingNoteContent}
                       onChange={this.props.childProps.onWriteNoteContent}
                />
                <div className="footer-container">
                    <Button name="close" className="footer-button" onClick={() => this.props.childProps.onCloseNewNote()}>Close</Button>
                </div>
            </Card>
        )
    }
};