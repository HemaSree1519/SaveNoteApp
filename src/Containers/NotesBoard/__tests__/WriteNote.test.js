import React from 'react';
import {shallow} from 'enzyme';
import EditNote from '../views/EditNote';
import WriteNote from "../views/WriteNote";

describe('EditNote', () => {
    let props;
    let component;
    beforeEach(() => {
        props = {
            writingNoteTitle: '',
            writingNoteContent: '',
            onCloseNewNote: jest.fn(),
            onWriteNoteContent: jest.fn(),
            onWriteNoteTitle: jest.fn(),
        };
        component = shallow(<WriteNote childProps={props}/>);
    });
    it('should render correctly', () => {
        expect(component).toMatchSnapshot();
    });
    it("should call title writing function when value changed", () => {
        component.find('Input[name="title"]').simulate('change', {value: 'title'});
        expect(props.onWriteNoteTitle).toHaveBeenCalledTimes(1);
    });
    it("should call content writing function when value changed", () => {
        component.find('Input[name="content"]').simulate('change', {value: 'content'});
        expect(props.onWriteNoteContent).toHaveBeenCalledTimes(1);
    });
    it("should call close function upon close button pressed", () => {
        component.find('Button[name="close"]').simulate('click');
        expect(props.onCloseNewNote).toHaveBeenCalledTimes(1);
    });
});