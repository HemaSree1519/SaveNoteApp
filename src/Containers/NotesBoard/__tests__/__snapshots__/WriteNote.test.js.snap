exports[`EditNote should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WriteNote
    childProps={
        Object {
            "onCloseNewNote": [Function],
            "onWriteNoteContent": [Function],
            "onWriteNoteTitle": [Function],
            "writingNoteContent": "",
            "writingNoteTitle": "",
          }
    } />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Input
          className="note-card-title-input"
          name="title"
          onChange={[Function]}
          placeholder="Title"
          type="text"
          value="" />,
        <Input
          className="note-card-content-input"
          name="content"
          onChange={[Function]}
          placeholder="Write a note..."
          type="textarea"
          value="" />,
        <div
          className="footer-container">
          <Button
                    className="footer-button"
                    color="secondary"
                    name="close"
                    onClick={[Function]}
                    tag="button">
                    Close
          </Button>
</div>,
      ],
      "className": "note-input",
      "tag": "div",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "note-card-title-input",
          "name": "title",
          "onChange": [Function],
          "placeholder": "Title",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "note-card-content-input",
          "name": "content",
          "onChange": [Function],
          "placeholder": "Write a note...",
          "type": "textarea",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Button
            className="footer-button"
            color="secondary"
            name="close"
            onClick={[Function]}
            tag="button">
            Close
</Button>,
          "className": "footer-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Close",
            "className": "footer-button",
            "color": "secondary",
            "name": "close",
            "onClick": [Function],
            "tag": "button",
          },
          "ref": null,
          "rendered": "Close",
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Input
            className="note-card-title-input"
            name="title"
            onChange={[Function]}
            placeholder="Title"
            type="text"
            value="" />,
          <Input
            className="note-card-content-input"
            name="content"
            onChange={[Function]}
            placeholder="Write a note..."
            type="textarea"
            value="" />,
          <div
            className="footer-container">
            <Button
                        className="footer-button"
                        color="secondary"
                        name="close"
                        onClick={[Function]}
                        tag="button">
                        Close
            </Button>
</div>,
        ],
        "className": "note-input",
        "tag": "div",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "note-card-title-input",
            "name": "title",
            "onChange": [Function],
            "placeholder": "Title",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "note-card-content-input",
            "name": "content",
            "onChange": [Function],
            "placeholder": "Write a note...",
            "type": "textarea",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Button
              className="footer-button"
              color="secondary"
              name="close"
              onClick={[Function]}
              tag="button">
              Close
</Button>,
            "className": "footer-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Close",
              "className": "footer-button",
              "color": "secondary",
              "name": "close",
              "onClick": [Function],
              "tag": "button",
            },
            "ref": null,
            "rendered": "Close",
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
